src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilterTest.java

```diff
@@ -410,21 +411,31 @@ public void testFindNearestMatchDontAllowSameColumn() {
     @Test
     public void testTagsAreClearedEachRun() {
         final SuppressionCommentFilter suppressionCommentFilter = new SuppressionCommentFilter();
-        final FileContents contents =
-                new FileContents("filename", "//CHECKSTYLE:OFF", "line2");
+        final List<?> tags1 = getTagsAfterExecution(suppressionCommentFilter,
+                "filename1", "//CHECKSTYLE:OFF", "line2");
+        assertEquals("Invalid tags size", 1, tags1.size());
+        final List<?> tags2 = getTagsAfterExecution(suppressionCommentFilter,
+                "filename2", "No comments in this file");
+        assertEquals("Invalid tags size", 0, tags2.size());
+    }
+
+    /**
+     * Calls the filter with a minimal set of inputs and returns a list of
+     * {@link SuppressionCommentFilter} internal type {@code Tag}.
+     * Our goal is 100% test coverage, for this we use white-box testing.
+     * So we need access to the implementation details. For this reason,
+     * it is necessary to use reflection to gain access to the inner field here.
+     *
+     * @return {@code Tag} list
+     */
+    private static List<?> getTagsAfterExecution(SuppressionCommentFilter filter,
```

Comment:
Shouldn't `testFindNearestMatchDontAllowSameColumn` use this method too?

Reply:
Surprisingly, this test is broken. It's name implies an invocation of `com.puppycrawl.tools.checkstyle.filters.SuppressionCommentFilter#findNearestMatch` which returns `null`, but it doesn't.

I'll fix this in a separate issue. And no, `getTagsAfterExecution` shouldn't be used here, since there must be a tag, but the tag should be filtered out by `findNearestMatch`.

Reply:
@pbludov https://github.com/checkstyle/checkstyle/commit/7d48c0689c68d1ef30993452abe6de0df09f4fd3#diff-969f42dc6092117e9bfa2dc4378d5a05R330
It was added for pitest. I would try removing it and see what pitest says the difference is.